### ISSUE TITLE
Data Exporting from a Data Entry Centre

### ISSUE THREAD

**Issue by JohnMwashuma** [onaio/tally-ho#223]
*Status: This functionality does not exist in the current software.*

All databases – remote and local - shall be designed in a manner which will allow them to be simply consolidated into a single file. A download data function shall be added and accessible from a screen which will allow the simple access to raw data from a data entry server, which can be uploaded into the system in another location.

The downloaded data should be easily loadable into a local version of the tally software, so that after the tabulation is completed, any subsequent edits can be made as per challenges and appeals.

**Comment by JohnMwashuma** [comment]
@pld I need your help understanding this requirement. Does this mean we need to build functionality for exporting the csv files that are normally imported when creating a tally? Or just functionality for exporting all the raw data from the database, which can then be used to set up another tally system.

**Comment by pld** [comment]
I think this issue is ambiguous. It was originally in the context of multiple remote tally systems and sounds like it's main point was you can load one entire system into another.

But, now that we're not doing that, it's unclear if it's important.

If it is important, we should say we have exports but if the request is for reloading, we can do this. If they're OK with a more technical person doing this, which would probably have to happen anyways since this is primary useful after a new app has been set up, there are db commands to accomplish this.

If that's not enough, we can give the options of (which you've already listed some of)

1. exporting the tally files in the same format you would import them in (do we not have this functionality?)
2. export the tally files but as one big combined file
3. export and import a database image (or something close to it) -- for this we need to know whether this is only for a single tally or for all tallies in a system
4. export a database but more human readable

I think the best option is to NOT do this, b/c of the point above that database dump and load tools cover it and are no more technical than what's already needed to set up a new server.

Marking this as blocked, @akahure and I can chat about it in the future and then check w/the client.

**Comment by akahure** [comment]
@pld @JohnMwashuma agreed, we should keep this as simple as possible. maybe only exporting the raw files in the system for setup but if they need anything over that (which includes options 2 and 3 )- we can engage them as a future piece/ or in the code walkthrough for the person they anticipate will be the 'Super Admin'. I do not think they will have someone technical to handle this, hence the support piece. 

For Option 1 we have this in place already
